src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -16,30 +17,28 @@ namespace Npgsql.TypeMapping
 {
     sealed class ConnectorTypeMapper : TypeMapperBase
     {
-        /// <summary>
-        /// The connector to which this type mapper belongs.
-        /// </summary>
         readonly NpgsqlConnector _connector;
 
         NpgsqlDatabaseInfo? _databaseInfo;
 
-        /// <summary>
-        /// Type information for the database of this mapper.
-        /// </summary>
         internal NpgsqlDatabaseInfo DatabaseInfo
-            => _databaseInfo ?? throw new InvalidOperationException("Internal error: this type mapper hasn't yet been bound to a database info object");
+        {
+            get => _databaseInfo ?? throw new InvalidOperationException("Internal error: this type mapper hasn't yet been bound to a database info object");
+            set => _databaseInfo = value;
```

Comment:
It doesn't look like we set it anywhere - maybe return it as it was before?

Reply:
This is set in NpgsqlConnector.LoadDatabaseInfo. It needs to be set late (after construction) since the type mapper is still necessary in order to do the type loading problem at the very first startup. How would you improve this?

Reply:
Ah, I see - before it was set through `Bind`, and now through a property. Then yeah, it should be OK.